pretyping/glob_ops.ml

```diff
@@ -61,83 +61,78 @@ let cast_type_eq eq t1 t2 = match t1, t2 with
 | CastNative t1, CastNative t2 -> eq t1 t2
 | _ -> false
 
-let rec glob_constr_eq c1 c2 = match c1, c2 with
+let rec mk_glob_constr_eq f c1 c2 = match c1, c2 with
```

Comment:
This `rec` is now only used to bring the subfunctions into scope. That confused me, I thought `mk_glob_constr_eq` was calling itself recursively on some subterms and using `f` on others in some occult way.
Could you reorder the functions to get rid of the `rec`, or pull from https://github.com/SkySkimmer/coq/tree/trunk+fix-5523-missing-support-match-recursive-notations where I did that?